### PR TITLE
Issue #3180 - Review client support for Unix sockets.

### PR DESCRIPTION
**Pull request by sbordet**
Reviewed the implementation.
Got rid of the single channel stored in the HttpClientTransport.
Re-enabled tests on the Unix socket transport.
Updated JNR to 0.22.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

**Comment by sbordet** [comment]
This PR aligns a bit better 9.4.x with 10.0.x - although 10.0.x will probably have some additional fixes/features.

**Comment by sbordet** [comment]
Test failures in this PR are most likely due to jnr/jnr-unixsocket#69.

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportProvider.java [hunk 1]
```diff
@@ -35,21 +35,16 @@
...
-        {
-            return Arrays.stream(transports.split("\\s*,\\s*"))
-                    .map(Transport::valueOf);
-        }
+            return Arrays.stream(transports.split("\\s*,\\s*")).map(Transport::valueOf);
 
-        // TODO #2014 too many test failures, don't test unix socket client for now.
-        // if (OS.IS_UNIX)
-        //     return Transport.values();
+         if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("linux"))
```
Comment (joakime) [inline tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportProvider.java]: There's an easier way to do this for test cases.

See example usage in https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-start/src/test/java/org/eclipse/jetty/start/PathMatchersSearchRootTest.java

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportScenario.java [hunk 2]
```diff
@@ -254,13 +247,13 @@ public HttpClientTransport provideClientTransport(Transport transport)
...
         }
-        return result.toArray(new ConnectionFactory[result.size()]);
+        return result.toArray(new ConnectionFactory[0]);
     }
 
     public void setConnectionIdleTimeout(long idleTimeout)
     {
         if (connector instanceof AbstractConnector)
-            AbstractConnector.class.cast(connector).setIdleTimeout(idleTimeout);
+            ((AbstractConnector)connector).setIdleTimeout(idleTimeout);
```
Comment (joakime) [inline tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportScenario.java]: Why this change?

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportProvider.java (hunk 1 above)
Comment (sbordet) [reply tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportProvider.java]: Thanks, done.

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportScenario.java (hunk 2 above)
Comment (sbordet) [reply tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportScenario.java]: To get rid of warnings.

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportProvider.java (hunk 1 above)
Comment (joakime) [reply tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/TransportProvider.java]: Am I missing a commit?  didn't see this changed.
